src/Common/tests/System/Collections/ICollection.NonGeneric.Tests.cs

```diff
@@ -64,6 +64,25 @@ protected virtual ICollection NonGenericICollectionFactory(int count)
         protected virtual Type ICollection_NonGeneric_CopyTo_ArrayOfIncorrectValueType_ThrowType => typeof(ArgumentException);
 
         /// <summary>
+        /// Used for the ICollection_NonGeneric_CopyTo_NonZeroLowerBound test where we try to call CopyTo
+        /// on an Array of with a non-zero lower bound.
+        /// Most implementations throw an ArgumentException, but others (e.g. Hashtable) throw
+        /// an IndexOutOfRangeException due to bugs in the implementation.
+        /// </summary>
+        protected virtual Type ICollection_NonGeneric_CopyTo_NonZeroLowerBound_ThrowType => typeof(ArgumentException);
+
+        /// <summary>
+        /// Used for the ICollection_NonGeneric_CopyTo_NonZeroLowerBound test where we try to call CopyTo
+        /// on an Array of with a non-zero lower bound.
+        /// Most implementations throw an ArgumentException, but others (e.g. Hashtable) throw
+        /// an IndexOutOfRangeException due to bugs in the implementation.
+        /// This works around an inconsistency in HybridDictionary: when HybridDictionary has a large count,
+        /// it uses an internal Hashtable, so there is an inconsistent result at different lengths. We should
+        /// therefore skip this test.
+        /// </summary>
+        protected virtual bool ICollection_NonGeneric_CopyTo_TestNonZeroLowerBound => true;
```

Comment:
If this is only ever false for one collection it'd probably be better to just make the test virtual and override it in HybridDictionary with a stub (or even better, a test that catches different exceptions for different sizes).